### PR TITLE
Owasp mobile top 10 2024

### PR DESCRIPTION
**Pull request by avzz-19**

### CODE REVIEW

File: tests/integration/components/file-details/vulnerability-analysis-details/index-test.js [hunk 1]
```diff
@@ -714,7 +708,9 @@ module(
       assert.dom('[data-test-analysisDetails-header]').exists();
 
       assert.dom('[data-test-analysisDetails-regulatory="cvssv3"]').exists();
-      assert.dom('[data-test-analysisDetails-regulatory="owasp"]').exists();
+      assert
```
Comment (future-pirate-king) [inline tests/integration/components/file-details/vulnerability-analysis-details/index-test.js:711]: add new line above

File: tests/unit/models/owaspmobile2024-test.js [hunk 2]
```diff
@@ -4,11 +4,13 @@ import { setupTest } from 'ember-qunit';
...
+module('Unit | Model | owaspmobile2024', function (hooks) {
   setupTest(hooks);
 
-  test('it exists', function(assert) {
-    const model = run(() => this.owner.lookup('service:store').createRecord('owasp'));
+  test('it exists', function (assert) {
+    const model = run(() =>
+      this.owner.lookup('service:store').createRecord('owaspmobile2024')
+    );
     assert.ok(!!model);
```
Comment (future-pirate-king) [inline tests/unit/models/owaspmobile2024-test.js:14]: add new line above

File: translations/en.json [hunk 3]
```diff
@@ -860,7 +860,7 @@
   "overriddenRisk": "Overridden Risk",
   "overrideRisk": "Override Risk",
   "overview": "Overview",
-  "owasp": "OWASP",
```
Comment (future-pirate-king) [inline translations/en.json:863]: we can keep this

File: mirage/models/owaspmobile2024.ts
Comment (future-pirate-king) [inline mirage/models/owaspmobile2024.ts:1]: do we require this model ?

File: app/components/file-details/vulnerability-analysis-details/index.ts [hunk 4]
```diff
@@ -45,11 +45,11 @@ export default class FileDetailsVulnerabilityAnalysisDetailsComponent extends Co
   get regulatoryContent() {
     return [
       {
-        key: 'owasp',
-        heading: this.intl.t('owasp'),
+        key: 'owaspmobile2024',
+        heading: this.intl.t('owaspmobile2024'),
         title: this.intl.t('owaspExpansion'),
```
Comment (future-pirate-king) [inline app/components/file-details/vulnerability-analysis-details/index.ts:50]: this should also be updated

File: app/adapters/owaspmobile2024.ts
Comment (future-pirate-king) [inline app/adapters/owaspmobile2024.ts:1]: need to remove old `owasp` adapter

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/adapters/owaspmobile2024.ts
Comment (avzz-19) [reply app/adapters/owaspmobile2024.ts:1]: There was no adapter for OWASP earlier.

File: mirage/models/owaspmobile2024.ts
Comment (avzz-19) [reply mirage/models/owaspmobile2024.ts:1]: yes, same as OWASP API TOP 10

Comment (avzz-19) [reply mirage/models/owaspmobile2024.ts:1]: removed

**Review by future-pirate-king — APPROVED** [review]

File: app/components/file-details/vulnerability-analysis-details/index.ts [hunk 5]
```diff
@@ -42,14 +42,26 @@ export default class FileDetailsVulnerabilityAnalysisDetailsComponent extends Co
...
-        hasContent: !isEmpty(this.analysis.owasp.toArray()),
+        hasContent:
+          !isEmpty(this.analysis.owasp.toArray()) && !this.isOwasp2024,
+      },
+      {
+        key: 'owaspmobile2024',
+        heading: this.intl.t('owaspmobile2024'),
+        title: this.intl.t('owaspMobile2024Expansion'),
+        contents: this.analysis.owaspmobile2024.toArray(),
+        hasContent: !isEmpty(this.analysis.owaspmobile2024.toArray()),
```
Comment (Yibaebi) [inline app/components/file-details/vulnerability-analysis-details/index.ts:64]: You can use the `isOwasp2024` getter here also.

File: mirage/factories/owaspmobile2024.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { Factory } from 'miragejs';
+import { faker } from '@faker-js/faker';
+
+export default Factory.extend({
```
Comment (Yibaebi) [inline mirage/factories/owaspmobile2024.ts:4]: Can the factory object be put in a variable? Similar to how the vulnerability factory is done. This is for the cypress tests.

File: tests/unit/models/owasp-test.js [hunk 7]
```diff
@@ -4,11 +4,13 @@ import { setupTest } from 'ember-qunit';
...
+module('Unit | Model | owasp', function (hooks) {
   setupTest(hooks);
 
-  test('it exists', function(assert) {
-    const model = run(() => this.owner.lookup('service:store').createRecord('owasp'));
+  test('it exists', function (assert) {
+    const model = run(() =>
+      this.owner.lookup('service:store').createRecord('owasp')
+    );
     assert.ok(!!model);
```
Comment (Yibaebi) [inline tests/unit/models/owasp-test.js:14]: Add new line above.

**Review by Yibaebi — CHANGES_REQUESTED** [review]

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/file-details/compliance-insights/index.hbs [hunk 8]
```diff
@@ -1,17 +1,33 @@
...
+
+  {{else}}
+    <AkChart
+      {{style maxWidth='350px'}}
+      @width='100%'
+      @height='200px'
+      @option={{this.owaspData.mobileOptions}}
+      data-test-complianceInsights-mobileChart
+    />
+    <AkChart
```
Comment (future-pirate-king) [inline app/components/file-details/compliance-insights/index.hbs:25]: add new line above

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Review by Yibaebi — APPROVED** [review]

**Pull request merged**